presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -180,9 +181,15 @@ public static double toUnixTimeFromTimestampWithTimeZone(@SqlType(StandardTypes.
     // the maximum year represented by 64bits timestamp is ~584944387 it may require up to 35 characters.
     public static Slice toISO8601FromTimestamp(ConnectorSession session, @SqlType(StandardTypes.TIMESTAMP) long timestamp)
     {
-        DateTimeFormatter formatter = ISODateTimeFormat.dateTime()
-                .withChronology(getChronology(session.getTimeZoneKey()));
-        return utf8Slice(formatter.print(timestamp));
+        if (session.isLegacyTimestamp()) {
+            DateTimeFormatter formatter = ISODateTimeFormat.dateTime()
+                    .withChronology(getChronology(session.getTimeZoneKey()));
+            return utf8Slice(formatter.print(timestamp));
+        }
+        else {
+            // FIXME: Simply remove this function when legacyTimestamp flag will be dropped
+            throw new PrestoException(FUNCTION_NOT_FOUND, "to_iso8601(timestamp) does not exist");
```

Comment:
Maybe https://en.wikipedia.org/wiki/ISO_8601 is not authoritative, but  quoting

> If no UTC relation information is given with a time representation, the time is assumed to be in local time.

So, there is ISO 8601 format suitable for timestamp, it just does not contain the tz info. Thus, an alternative to disabling the method would be changing its semantics.